src/Symfony/Component/RateLimiter/RateLimiterBuilder.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+use Symfony\Component\Lock\LockInterface;
+use Symfony\Component\RateLimiter\Policy\FixedWindowLimiter;
+use Symfony\Component\RateLimiter\Policy\NoLimiter;
+use Symfony\Component\RateLimiter\Policy\Rate;
+use Symfony\Component\RateLimiter\Policy\SlidingWindowLimiter;
+use Symfony\Component\RateLimiter\Policy\TokenBucketLimiter;
+use Symfony\Component\RateLimiter\Storage\StorageInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+class RateLimiterBuilder
+{
+    public function __construct(
+        private StorageInterface $storage,
+        private ?LockInterface $lock = null,
+    ) {
+    }
+
+    /**
+     * @param string               $id       Unique identifier for the rate limiter
+     * @param int                  $limit    Maximum allowed hits for the passed interval
+     * @param \DateInterval|string $interval If string, must be a number followed by "second",
+     *                                       "minute", "hour", "day", "week" or "month" (or their
+     *                                       plural equivalent)
```

Comment:
I think that we tend to write something like "Any valid date interval string" and maybe link to the right PHP doc page?

Reply:
I copied the note from here: https://github.com/symfony/symfony/blob/561d4a086c13bde50a16e0f5408582d2e1d85897/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L2526

It's a little custom as we don't want the + included

Reply:
Alright then 🙂